### ISSUE TITLE
Add option to auto-complete PRs in VS to the insertion tool

### ISSUE THREAD

**Issue by dmonroym** [dotnet/roslyn-tools#697]
Would be nice to have an option to allow a merge from GitHub PR to automatically trigger everything including an automatic insertion

**Comment by dmonroym** [comment]
Currently working on something based on the option set here: https://devdiv.visualstudio.com/DevDiv/_git/PostBuildSteps#path=%2Fsrc%2FSubmitPullRequest%2FProgram.cs&version=GBmaster&_a=contents

**Comment by RikkiGibson** [comment]
this feature is already making life a lot nicer, thanks @dmonroym. Something just came to mind to me about it though--is there any possibility this will result in PRs being completed out-of-order, e.g.

1. the PR for build 2 is auto-completed
2. then the checks for the PR for build 1 complete later due to some infra issue, and it is still marked auto-complete

This still might be fine, because the PR for build 1 may get blocked by merge conflicts in the base branch. But I wanted to make sure.

**Comment by dmonroym** [comment]
There could be potential issues but not with it auto-completing because, as you mentioned, there would be merge conflicts which would have to be manually resolved.

The main issue though is if someone doesn't notice there was an insertion of build 2, resolved build 1, and now build 1 is the most recent.
There are two suggestions I've been thinking about but would like more feedback before:
	1. It would be nice to notify the team immediately after auto-complete failed so either someone re-runs (which could also be automated?) it or they check why it failed.
	2. When there's a newer insertion we should abandon the previous one to avoid confusion (albeit this in itself can cause confusion if you're trying to insert feature A before a deadline but not feature B and the insertion for A keeps failing)

**Comment by RikkiGibson** [comment]
don't auto-abandon, but consider disabling auto-complete for "older" PRs at the time the newer PR is created, if the auto-complete flag has been passed.

**Comment by RikkiGibson** [comment]
This also needs to be done on a per-branch basis. e.g. search for VS PRs with the same base branch and the same component name.

**Comment by dmonroym** [comment]
> don't auto-abandon, but consider disabling auto-complete for "older" PRs at the time the newer PR is created, if the auto-complete flag has been passed.
Good suggestion, it might be worth creating a comment tagging the right team to make them aware as well.
